tests/trace_plugin_test.py

```diff
@@ -0,0 +1,91 @@
+        self.walletMgr.killall(allInstances=True)
+        self.walletMgr.cleanup()
+
+    # start keosd and nodeos
+    def startEnv(self) :
+        account_names = ["alice", "bob", "charlie"]
+        traceNodeosArgs = " --plugin eosio::trace_api_plugin --trace-no-abis --trace-dir=."
+        self.cluster.launch(totalNodes=1, extraNodeosArgs=traceNodeosArgs)
+        self.walletMgr.launch()
+        testWalletName="testwallet"
+        testWallet=self.walletMgr.create(testWalletName, [self.cluster.eosioAccount, self.cluster.defproduceraAccount])
+        self.cluster.validateAccounts(None)
+        self.accounts=Cluster.createAccountKeys(len(account_names))
+        node = self.cluster.getNode(0)
+        for idx in range(len(account_names)):
+            self.accounts[idx].name =  account_names[idx]
+            self.walletMgr.importKey(self.accounts[idx], testWallet)
+        for account in self.accounts:
+            node.createInitializeAccount(account, self.cluster.eosioAccount, buyRAM=1000000, stakedDeposit=5000000, waitForTransBlock=True, exitOnError=True)
+        time.sleep(self.sleep_s)
+
+    def get_block(self, params: str, node: Node) -> json:
+        time.sleep(self.sleep_s)
+        base_cmd_str = ("curl http://%s:%s/v1/") % (TestHelper.LOCAL_HOST, node.port)
+        cmd_str = base_cmd_str + "trace_api/get_block  -X POST -d " + ("'{\"block_num\":%s}'") % params
+        return Utils.runCmdReturnJson(cmd_str)
+
+    def test_TraceApi(self) :
+        node = self.cluster.getNode(0)
+        for account in self.accounts:
+            self.assertIsNotNone(node.verifyAccount(account))
+
+        expectedAmount = Node.currencyIntToStr(5000000, CORE_SYMBOL)
+        account_balances = []
+        for account in self.accounts:
+            amount = node.getAccountEosBalanceStr(account.name)
+            self.assertEqual(amount, expectedAmount)
+            account_balances.append(amount)
+
+        xferAmount = Node.currencyIntToStr(123456, CORE_SYMBOL)
+        trans = node.transferFunds(self.accounts[0], self.accounts[1], xferAmount, "test transfer a->b")
+        transId = Node.getTransId(trans)
+        blockNum = Node.getTransBlockNum(trans)
+
+        self.assertEqual(node.getAccountEosBalanceStr(self.accounts[0].name), Utils.deduceAmount(expectedAmount, xferAmount))
+        self.assertEqual(node.getAccountEosBalanceStr(self.accounts[1].name), Utils.addAmount(expectedAmount, xferAmount))
+        ret_json = self.get_block(blockNum, node)
+        print(ret_json)
+        self.assertIn("transactions", ret_json)
+
+        isTrxInBlock = False
+        for trx in ret_json["transactions"]:
+            self.assertIn("id", trx)
+            if (trx["id"] == transId) :
+                isTrxInBlock = True
+        self.assertTrue(isTrxInBlock)
+
+    @classmethod
+    def setUpClass(self):
+        self.cleanEnv(self)
+        self.startEnv(self)
+
+    @classmethod
+    def tearDownClass(self):
+        self.cleanEnv(self)
```

Comment:
either need to check if test passed or just don't call cleanup() when tearing down